code/game/antagonist/station/traitor.dm

```diff
@@ -13,6 +13,13 @@ var/datum/antagonist/traitor/traitors
 	..()
 	traitors = src
 
+/datum/antagonist/traitor/build_candidate_list()
+	..()
+	for(var/datum/mind/player in candidates)
+		if(player.assigned_role && player.assigned_role == "AI")
```

Comment:
...can't you just add AI to the job blacklist for traitor?

Reply:
Good point